### PR TITLE
Upgrade caasp without scc

### PR DESCRIPTION
**Pull request by jsuchome**
The alternative to https://github.com/SUSE-Cloud/socok8s/pull/163

### CODE REVIEW

**Comment by Itxaka** [comment]
isnt this missing the velum part like https://github.com/SUSE-Cloud/socok8s/pull/163 ?

We should be able to do the same as there for the update path but exchange the suseconnect by the manual adding of the repos no?

**Comment by jsuchome** [comment]
> isnt this missing the velum part like #163 ?

Yep. See latest commit

**Comment by Itxaka** [comment]
This looks good to me, hopefully it works!

File: playbooks/openstack-prepare_caasp_workers_for_update.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      zypper_repository:
+        name: "{{ item }}"
+        repo: "{{ deploy_on_openstack_repos_to_configure[item] }}"
+        autorefresh: True
+        auto_import_keys: yes
+        state: present
+      loop: "{{ deploy_on_openstack_caasp_repos_per_imagename[deploy_on_openstack_caasp_image] }}"
+
+    - name: Perform dist-upgrade
+      command: transactional-update cleanup dup salt
```
Comment (jhesketh) [inline playbooks/openstack-prepare_caasp_workers_for_update.yml]: What happens if you do `transactional-update cleanup dup salt reboot`[0]?
Would that restart the servers when appropriate without needing to interact with velum?

[0] http://users.suse.com/~kukuk/SUSE-CaaSP-Docu/transactional-update.8.html#reboot

File: playbooks/openstack-update_caasp.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    - name: Load variables
+      include_vars: "{{ item }}"
+      loop:
+        - "{{ playbook_dir }}/../vars/common-vars.yml"
+        - "{{ playbook_dir }}/../vars/manifest.yml"
+      tags:
+        - always
+
+    - name: Use Velum tooling to update CaaSP admin node
+      shell: "./velum-interactions --update-admin --update-minions -e {{ upstream_repos_clone_folder }}/ccp/environment.json"
```
Comment (jhesketh) [inline playbooks/openstack-update_caasp.yml]: Do the `--update..` flags do the the `dup`? If so we wouldn't need to run it manually.

File: playbooks/openstack-update_caasp.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      args:
+        executable: /bin/bash
+        chdir: "{{ upstream_repos_clone_folder }}/ccp/velum-bootstrap"
+      tags:
+        - skip_ansible_lint
+
+    - name: Use Velum tooling to update CaaSP nodes
+      shell: "./velum-interactions --update-minions -e {{ upstream_repos_clone_folder }}/ccp/environment.json"
```
Comment (jhesketh) [inline playbooks/openstack-update_caasp.yml]: You have `--update-minions` in the previous task as well. Is this needed?

Comment (jsuchome) [reply playbooks/openstack-update_caasp.yml]: Ah, this seems like some forgotten commit

File: playbooks/openstack-prepare_caasp_workers_for_update.yml (hunk 1 above)
Comment (jsuchome) [reply playbooks/openstack-prepare_caasp_workers_for_update.yml]: My understanding was that the reboot should be done through the velum-interactions

**Comment by jsuchome** [comment]
Unfortunately it fails with https://bugzilla.suse.com/show_bug.cgi?id=1130032

**Comment by jsuchome** [comment]
@jhesketh I've changed repositories, tested it with somewhat mixed results. Maybe I was just impatient and it would work after all

**Comment by jsuchome** [comment]
@jhesketh ok, this finally seems to work for me. I've added little piece before the admin update, as was adviced by CaaSP people

File: playbooks/openstack-update_caasp.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        - "{{ playbook_dir }}/../vars/common-vars.yml"
+        - "{{ playbook_dir }}/../vars/manifest.yml"
+      tags:
+        - always
+
+    - name: Use Velum tooling to update CaaSP admin node
+      shell: "./velum-interactions --update-admin -e {{ upstream_repos_clone_folder }}/ccp/environment.json"
+      args:
+        executable: /bin/bash
+        chdir: "{{ upstream_repos_clone_folder }}/ccp/velum-bootstrap"
```
Comment (Itxaka) [inline playbooks/openstack-update_caasp.yml]: Does this command waits for the admin to be restarted? or it just triggers the restart and exits?

File: playbooks/openstack-update-caasp-salt-grains.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+#
+
+---
+- hosts: caasp-admin
+  gather_facts: no
+  tasks:
+    - name: Update the grains' states
+      shell: |
+        set -eo pipefail
+        docker exec -i $(docker ps | grep salt-master | awk '{print $1}') salt '*' saltutil.refresh_grains
```
Comment (Itxaka) [inline playbooks/openstack-update-caasp-salt-grains.yml]: nice!

File: playbooks/openstack-update_caasp.yml (hunk 4 above)
Comment (jsuchome) [reply playbooks/openstack-update_caasp.yml]: I think it waits for admin to be back online

**Comment by jsuchome** [comment]
OK, merging

**Pull request merged**